File: Java/SimpleReadFromFile.java
/*
 * This Java program reads from a text file, line by line, and 
 * prints the file contents to stdout, line by line
 * 
 * Created By: Josh Johnson
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class SimpleReadFromFile {
	public static void main(String args[]) {
		try {			
			File file = new File("../Test_Files/Poem.txt");
			Scanner reader = new Scanner(file);
			
			while (reader.hasNextLine()) {
				String line = reader.nextLine();
				System.out.print(line);

				/* 
				 * this condition removes the extra line after the 
				 * last line of text in the file 
				 */
				if (reader.hasNextLine() == true) {
					System.out.println();
				}
			}
						
			reader.close();
		} catch (FileNotFoundException e) {
			System.err.println("No file has been found");
			e.printStackTrace();
		}
	}
}

File: C/Simple_Read_From_File.c
/*
 * This C program reads from a text file, character by character, and 
 * prints the file contents to stdout, character by character
 * 
 * Created By: Josh Johnson
 */

#include <stdio.h>
#include <stdlib.h>

int main() {
    FILE* file;
    char c;

    /* tries to open file */
    if ((file = fopen("../Test_Files/Basic_Text.txt", "r")) == NULL) {
        printf("Could not open the file");

        exit(EXIT_FAILURE); /* same as exit(1) */
    }

    /* prints file content character by character to output stream */
    do {
        c = fgetc(file);

        if (!feof(file)) {
            printf("%c", c);
        }
    } while (!feof(file));

    fclose(file);
    return EXIT_SUCCESS; /* Same as return 0 */
}
